Config faker server

mchuong1 opened the pull request.

mchuong1 committed:
config url and server port

mchuong1 committed:
Config variable add REACT_APP_

mchuong1 committed:
config protocol

mchuong1 committed:
Take out hard coded colon in url

mchuong1 committed:
update constants

mchuong1 committed:
Update Urls.js

mchuong1 committed:
Update AccountProfile.js

mchuong1 committed:
Update Tools.js

mchuong1 committed:
Ironout Account Profile for generic Get

mchuong1 committed:
Remove unused datasource Variable

1. Add key property to li element tag
2. add key property to div tag
3. turn if block to tertionary

mchuong1 committed:
Update data to reflect the same structure in MongoDB

1. remove users url
2.  remove advance search
3.  remove unused import from AccountProfile.js

mchuong1 committed:
Merge branch 'master' into config-faker-server

mchuong1 committed:
Have UserProfile use data from MongoDB

mchuong1 committed:
Add Params for user endpoint

1. Update mockData and added userId and JobId
2. Remove extra LoginSignup
3.  Add in test url for local backend
4.  In card component, pass userId and JobId in url route

mchuong1 commented:
getuserById has not been configured yet for homepage

mchuong1 commented:
The corresponding account page will pull up for the user card on search results

mchuong1 committed:
Merge branch 'master' into config-faker-server

mchuong1 committed:
Add in userId as param to navigate to account profile

mchuong1 committed:
Update how parameters are retrieved from url

-NOTE: This only works on modern browsers. this will fail on Internet Explorer

mchuong1 committed:
Update method to get all params

mchuong1 committed:
Append extra params with brackets

t-schell commented on src/data/DataManager.js:
```diff
@@ -15,45 +15,49 @@ export default class DataManager extends Component {
         let dataLoad = []
         for (var index in paths) {
-            dataLoad.push(dataSource.GetData(paths[index]))
+            paths[index] == 'user' ? dataLoad.push(dataSource.GetData(paths[index], param)) : dataLoad.push(dataSource.GetData(paths[index]))
```
Maybe could be generalized to work for all endpoints with a param instead of just user. Maybe the load data function could accept a list of objects that hold the path and it's associated params.

mchuong1 replied:
This is a good point, I definitely need to revisit this section of code.

mchuong1 replied:
for this endpoint for account profile. I think its better narrow the endpoints to 1 rather than 3

t-schell commented on src/data/DataManager.js:
```diff
@@ -15,45 +15,49 @@ export default class DataManager extends Component {
     }
 
+    componentDidMount = () => {
+        let param = window.location.pathname.substr(16, 25)
```
This substring seems dependent on this one endpoint. Can we pass the userId param from the router in as props of the DataManager?

mchuong1 committed:
Update param name for card

-Update logic inside load data to only use params assigned to the endpoint name
-Update how parameters are separated in getData

mchuong1 committed:
Add return statements to return promises

t-schell reviewed (approved):
Looks good
